storage/storage_integration_tests/storage_http_tests.cpp

```diff
@@ -51,49 +70,119 @@ UNIT_TEST(StorageDownloadNodeAndDeleteNodeTests)
     }
   };
 
-  auto ProgressFunction = [&storage](TCountryId const & countryId, TLocalAndRemoteSize const & mapSize)
+  storage.Init(didDownload, [](TCountryId const &, Storage::TLocalFilePtr const){return false;});
+  storage.RegisterAllLocalMaps();
+  storage.Subscribe(changeCountryFunction, progress);
+  storage.SetDownloadingUrlsForTesting({kTestWebServer});
+}
+
+class StorageHttpTest
+{
+  WritableDirChanger const m_writableDirChanger;
+  Storage m_storage;
+  string const m_version;
+  tests_support::ScopedDir const m_cleanupVersionDir;
+
+public:
+  StorageHttpTest()
+    : m_writableDirChanger(kMapTestDir), m_storage(COUNTRIES_FILE),
+      m_version(strings::to_string(m_storage.GetCurrentDataVersion())),
+      m_cleanupVersionDir(m_version)
+  {
+    TEST(version::IsSingleMwm(m_storage.GetCurrentDataVersion()), ());
+  }
+
+  Storage & GetStorage() { return m_storage; }
+  string const & GetVersion() const { return m_version; }
+};
+}  // namespace
+
+UNIT_CLASS_TEST(StorageHttpTest, StorageDownloadNodeAndDeleteNode)
+{
+  auto const progressFunction = [this](TCountryId const & countryId, TLocalAndRemoteSize const & mapSize)
   {
     NodeAttrs nodeAttrs;
-    storage.GetNodeAttrs(countryId, nodeAttrs);
+    GetStorage().GetNodeAttrs(countryId, nodeAttrs);
 
     TEST_EQUAL(mapSize.first, nodeAttrs.m_downloadingProgress.first, (countryId));
     TEST_EQUAL(mapSize.second, nodeAttrs.m_downloadingProgress.second, (countryId));
     TEST_EQUAL(countryId, kCountryId, (countryId));
   };
 
-  storage.Init(Update, [](TCountryId const &, storage::Storage::TLocalFilePtr const){return false;});
-  storage.RegisterAllLocalMaps();
-  storage.Subscribe(ChangeCountryFunction, ProgressFunction);
-  storage.SetDownloadingUrlsForTesting({kTestWebServer});
-  string const version = strings::to_string(storage.GetCurrentDataVersion());
-  tests_support::ScopedDir cleanupVersionDir(version);
-
-  string const mwmFullPath = my::JoinFoldersToPath({GetPlatform().WritableDir(), version},
-                                                   kCountryId + DATA_FILE_EXTENSION);
-  string const downloadingFullPath = my::JoinFoldersToPath(
-      {GetPlatform().WritableDir(), version},
-      kCountryId + DATA_FILE_EXTENSION READY_FILE_EXTENSION DOWNLOADING_FILE_EXTENSION);
-  string const resumeFullPath = my::JoinFoldersToPath(
-      {GetPlatform().WritableDir(), version},
-      kCountryId + DATA_FILE_EXTENSION READY_FILE_EXTENSION RESUME_FILE_EXTENSION);
-  Platform & platform = GetPlatform();
+  InitStorage(GetStorage(), Update, progressFunction);
+
+  string const mwmFullPath = GetMwmFullPath(kCountryId, GetVersion());
+  string const downloadingFullPath = GetDownloadingFullPath(kCountryId, GetVersion());
+  string const resumeFullPath = GetResumeFullPath(kCountryId, GetVersion());
 
   // Downloading to an empty directory.
-  storage.DownloadNode(kCountryId);
+  GetStorage().DownloadNode(kCountryId);
```

Comment:
You don't need GetStorage()/SetStorage() API. Just make m_storage protected and use it here as is.